### PR TITLE
adds required arguments to imagetag #979

### PR DESCRIPTION
**Pull request by chapmandu**

### CODE REVIEW

**Comment by neokoenig** [comment]
@dbelanger It should be true, as otherwise it breaks existing behaviour... (even if that behaviour is really annoying)

**Comment by chapmandu** [comment]
If you chaps are happy with this I'll merge it into master

**Comment by neokoenig** [comment]
It's a go from me

File: wheels/tests/view/assets/imagetag.cfc [hunk 1]
```diff
@@ -111,4 +111,25 @@ component extends="wheels.tests.Test" {
 		assert("e eq r");
 	}
 
+	function test_raises_exception_on_missing_image() {
+		path = "missing.jpg";
+		actual = raised("_controller.imageTag(source=path)");
+		expected = "Wheels.ImageFileNotFound";
+		assert("actual == expected");
```
Comment (chapmandu) [inline wheels/tests/view/assets/imagetag.cfc]:
```suggestion
		assert("actual eq expected");
```

File: wheels/tests/view/assets/imagetag.cfc [hunk 2]
```diff
@@ -111,4 +111,25 @@ component extends="wheels.tests.Test" {
...
+	}
+
+	function test_raises_exception_on_unsupported_image() {
+		path = "../wheels/tests/_assets/files/cfwheels-logo.txt";
+		actual = raised("_controller.imageTag(source=path)");
+		expected = "Wheels.ImageFormatNotSupported";
+		assert("actual == expected");
```
Comment (chapmandu) [inline wheels/tests/view/assets/imagetag.cfc]:
```suggestion
		assert("actual eq expected");
```

File: wheels/tests/view/assets/imagetag.cfc [hunk 3]
```diff
@@ -111,4 +111,25 @@ component extends="wheels.tests.Test" {
...
+	}
+
+	function test_accepts_missing_image() {
+		path = "missing.jpg";
+		actual = _controller.imageTag(source=path, required=false, alt="This may be broken");
+		expected = '<img alt="This may be broken" src="#imagePath#/#path#">';
+		assert("actual == expected");
```
Comment (chapmandu) [inline wheels/tests/view/assets/imagetag.cfc]:
```suggestion
		assert("actual eq expected");
```

**Comment by dbelanger** [comment]
Technically this change is functionality correct but I'm still not convinced default to "true" was the way to go.  While we protect existing behaviour, that existed behaviour isn't what's supposed to happen at the browser level.   Required = True is a good thing to have as it will force an error and we're being clear about that but if the standard is to return a silent 404 on the browser when the image file isn't present, I personally believe it should just return the dead link.

**Comment by neokoenig** [comment]
As a compromise, how about we commit the change in 2.2, but mark it as pending change in 2.3? i.e, like a deprecation notice (but not) ?

**Pull request merged**